### ISSUE TITLE
[Feature] Informing the algorithm it's in resume state

### ISSUE THREAD

**Issue by dreamflasher** [wandb/wandb#2236]
**Is your feature request related to a problem? Please describe.**
We are working a lot with preemptive instances, so we need a way to checkpoint and resume runs from checkpoints.

**Describe the solution you'd like**
Thank you for implementing resumable sweep runs: https://docs.wandb.ai/guides/track/advanced/resuming#preemptible-sweeps
Eventually we want this: ```As of version 0.10.31, requeued preempted runs begin execution from their initial step, rather than starting from the training iteration that they were at when they were preempted. Automated checkpointing for preempted runs is an area of future development.```

The first step to support this would be: https://github.com/wandb/client/issues/2234
And another parallel step would be this feature I am describing here.

When the algorithm is started from preempted state it should get a notification about this, ideally including the step count. This should be added as a command line argument to the train argument call.

**Comment by raubitsj** [comment]
@dreamflasher I think preemptible sweeps does support resuming right now (We are still testing this and will update the documentation when it is verified)

So this means that you end up resuming from the last step that was successfully logged.   What I think you need is a way to "rewind" back to the step of your last checkpoint.

For now I think you can emulate this behavior by logging your own step metric:
wandb.log(dict(mystep=1, acc=accuracy))

Then when you resume, you start logging early values of mystep.   The graphing will likely still show a discontinuity until we add proper support.

The plan is to refine add this "rewind" (likely to be renamed) functionality as proposed in this experimental PR:
https://github.com/wandb/client/pull/2079

**Comment by dreamflasher** [comment]
Thank you, that's already very helpful. But how would the algorithm know that it should start at a certain step count?

**Comment by raubitsj** [comment]
I was thinking it would be something like this:
```python
run = wandb.init(resume="allow")
if run.resumed:
     run.rewind(step=run.summary["last_checkpoint_step"])
     model = keras.models.load_model(wandb.restore("model-best.h5").name)
```

**Comment by raubitsj** [comment]
Everything but `run.rewind()` should work now.  

So the workaround is without rewind() is:
```
run = wandb.init(resume="allow")

# optional call which will make sure that charts for this metric are correlated to global_step (instead of the internal step counter)
run.define_metric("accuracy", step_metric="global_step")

global_step = 0

if run.resumed and (global_step := run.summary.get("last_checkpoint_global_step", 0):
     model = keras.models.load_model(wandb.restore("model-best.h5").name)
...

run.log(dict(global_step=(global_step := global_step + 1), accuracy=accuracy))
...
if should_save_checkpoint:
   run.save("model-best.h5", model)
   run.summary["last_checkpoint_global_step"] = global_step
```

**Comment by dreamflasher** [comment]
Thank you very much! That's amazing, I am glad to see that it's basically already possible.

**Comment by dreamflasher** [comment]
@bcsherma did you fix this?

**Comment by bcsherma** [comment]
@dannygoldstein has done quite a bit of work in this area, I don't believe the experimental PR is going to be merged but a more mature "rewind" feature will be available soon
